### PR TITLE
Tests: convert quick tests into regular tests

### PR DESCRIPTION
**Pull request by tamiko**
This has irked me quite a bit: our quick tests duplicate a lot of
functionality that is already available in the testsuite proper. So,
instead of reinventing the wheel, let us simply use the testsuite itself.

The following pull request:

 - refactors all quick tests into regular tests that are configured for all
   available build modes.

 - modifies the `test` target to properly depend on the library
   and the setup_tests_quick_tests target. This ensures that when calling
   `make test` or `ninja test` the library is compiled and quick tests are
   properly configured.

 - the `test` target will now execute `run_quick_tests.cmake` which does
   little more than invoking

     ctest -JN -R "quick_tests/(step.debug|step.release|affinity|.*.${_build_type})"

   which mimicks the old behavior, meaning, we only run quick tests in one
   configuration with the exception of the step and affinity tests.

Requires #14559

### CODE REVIEW

File: tests/run_quick_tests.cmake [hunk 1]
```diff
@@ -13,80 +13,80 @@
...
     endif()
-
   endforeach()
-
-  # The CMake command message(SEND_ERROR ...) is, to the best of the authors'
-  # knowledge, the only way to set the exit status of CMake to a nonzero value.
-  # If we used message(SEND_ERROR ...) at the top (with the actual error
-  # message) then subsequent messages (i.e., the test specific help) would not
-  # be printed. Hence, do it down here.
-  message(SEND_ERROR "")
```
Comment (tamiko) [inline tests/run_quick_tests.cmake:90]: This workaround has been unnecessary for a really long time...

**Comment by tamiko** [comment]
```
tamiko@peregrine dealii/build % ninja test          
[2/2] Running quicktests...
-- Running quick_tests in DEBUG mode with -j16:
Test project /home/tamiko/workspace/dealii/build
      Start  9: quick_tests/step.debug
      Start 13: quick_tests/umfpack.debug
      Start  5: quick_tests/gmsh.debug
      Start 10: quick_tests/sundials-ida.debug
      Start  4: quick_tests/boost_zlib.debug
      Start  3: quick_tests/assimp.debug
      Start  8: quick_tests/lapack.debug
      Start  2: quick_tests/affinity.debug
      Start 12: quick_tests/tbb.debug
      Start  7: quick_tests/kokkos.debug
      Start 11: quick_tests/symengine.debug
      Start  6: quick_tests/gsl.debug
      Start  1: quick_tests/adolc.debug
 1/13 Test  #7: quick_tests/kokkos.debug .........   Passed    0.07 sec
 2/13 Test  #1: quick_tests/adolc.debug ..........   Passed    0.08 sec
 3/13 Test #12: quick_tests/tbb.debug ............   Passed    0.09 sec
 4/13 Test  #6: quick_tests/gsl.debug ............   Passed    0.09 sec
 5/13 Test #11: quick_tests/symengine.debug ......   Passed    0.09 sec
 6/13 Test  #5: quick_tests/gmsh.debug ...........   Passed    0.10 sec
 7/13 Test  #4: quick_tests/boost_zlib.debug .....   Passed    0.10 sec
 8/13 Test  #2: quick_tests/affinity.debug .......   Passed    0.10 sec
 9/13 Test  #3: quick_tests/assimp.debug .........   Passed    0.10 sec
10/13 Test #13: quick_tests/umfpack.debug ........   Passed    0.11 sec
11/13 Test  #8: quick_tests/lapack.debug .........   Passed    0.10 sec
12/13 Test #10: quick_tests/sundials-ida.debug ...   Passed    0.11 sec
13/13 Test  #9: quick_tests/step.debug ...........***Failed    6.65 sec

92% tests passed, 1 tests failed out of 13

Total Test time (real) =   6.65 sec

The following tests FAILED:
          9 - quick_tests/step.debug (Failed)
Errors while running CTest
Output from these tests are in: /home/tamiko/workspace/dealii/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.


**************************************************************************
**                                                                      **
**                Error: Some of the quick tests failed.                **
**                                                                      **
**************************************************************************

Check the files quick_tests.log and Testing/Temporary/LastTest.log located
in your build directory for error messages. Alternatively, you can run all
failing quick tests again via

        $ ctest --rerun-failed --output-on-failure -R quick_tests/

If you are unable to fix this problem, write to the mailing list linked at
https://www.dealii.org
```

**Comment by tamiko** [comment]
*oops* I garbled the force push. Reponening

**Comment by tamiko** [comment]
In reference to https://github.com/dealii/dealii/issues/1921

**Comment by masterleinad** [comment]
I guess this needs #14559?

**Comment by tamiko** [comment]
@masterleinad Yes, this requires https://github.com/dealii/dealii/pull/14559

**Comment by tamiko** [comment]
@masterleinad @peterrum Alright. This one is now ready for review (and ready to pass the testsuite ^^).

**Review by masterleinad — APPROVED** [review]

**Pull request merged**